cmd/gcp-controller-manager/csr_signer.go

```diff
@@ -44,17 +43,23 @@ var (
 // ClusterSigningGKERetryBackoff is the backoff between GKE cluster signing retries.
 const ClusterSigningGKERetryBackoff = 500 * time.Millisecond
 
+type csrClient interface {
+	UpdateStatus(context.Context, *capi.CertificateSigningRequest, metav1.UpdateOptions) (*capi.CertificateSigningRequest, error)
+}
+
 // gkeSigner uses external calls to GKE in order to sign certificate signing
 // requests.
 type gkeSigner struct {
 	webhook      *webhook.GenericWebhook
 	ctx          *controllerContext
 	retryBackoff time.Duration
 	validators   []csrValidator
+
+	csrs csrClient
 }
 
 // newGKESigner will create a new instance of a gkeSigner.
-func newGKESigner(ctx *controllerContext) (*gkeSigner, error) {
+func newGKESigner(ctx *controllerContext, csrs csrClient) (*gkeSigner, error) {
```

Comment:
this is confusing, since ctx also contains a client interface... prefer using the pre-existing client

Reply:
Same concern as above --- I'm not sure how to maintain testability here.

Reply:
Resolved as above.